### ISSUE TITLE
Issues in C++ Implementation.

### ISSUE THREAD

**Issue by krs-world** [krs-world/bridges-cxx#34]
The current implementation is C++11 compliant. Most folks should have a C++11
compiler, I would imagine these days. Maybe turn some options. Its simply a lot more powerful with some of teh newer constructs with containers and makes the code a lot more elegant.

I have gone through with the first round of fixes. Need to do a little more testing
on the examples I have before pushing a version.   A few issues (fairly minor)
are if we want to convert some of the classes to static since you dont need
multiple instances of them (Bridges, Connector, Validation). 

And some issues I need to update my C++ knowledge on!

Great job, Dakota on all the little things you have caught on from my summer
project!

I will push a version of the C++ code tomorrow.

**Comment by dcarmer** [comment]
Okay, I've completed my round of updates and will be pushing them momentarily.

**Disclaimer:** I have not had time to perform a thorough test of these updates, so I just about guarantee there are still problems. Either way there are still things that need to be added.

I will be posting an issue for each file, to help document the changes I've made more thoroughly, but I'll discuss some of the bigger changes here.

To start I've:<ul>
<li>Removed the GraphList file, as it was a helper class that wasn't being utilized in the graphs.</li>
<li>Moved the Edge class into GraphAdjList, because its a helper class only used here.</li>
<li>Removed ADTVisualizer, and moved its JSON creation functions into each pertaining class.</li>
<li>Removed Connector, as its only methods were to change the server url(shouldn't be done), get the url(shouldn't be needed) and to post, I put post int bridges itself</li>
<li>Removed ArrayList, ArrayElement and Array</li>
<li>Moved ElementVisualizer into Element itself, as it seemed unnecessary to split them up</li>
<li>Added Color, a class to represent color values and handle their manipulation</li>
</ul>

**Comment by dcarmer** [comment]
I have finished making the additional changes. I have also done basic testing for it and it seems to be working for expected cases(still working on the more comprehensive tests).


I made several changes, but i noticed that the JSON being sent to the server repeats alot of info It sent in the previous post. While I realize there is no way to guarantee multiple calls occur, and thereby only send it once, I still feel this could be improved. 

For example the visualizer type is sent each post, like tree or linked list. This only effects the display on the "my projects" page and has no impact on the actual visualization. To this end, I would recommend only sending this data upon the first post per assignment. Furthermore this seems to be at odds with the fact that there is no check in place(I dont know if there is supposed to be or not) to prevent different data structures being sent to the same assignment, so 1.00 might be a tree, but 1.01 could be a graph. In which case this system doesnt make sense. Perhaps we should provide a variable for student organization such as **assignment title** for display on the projects tab instead?
